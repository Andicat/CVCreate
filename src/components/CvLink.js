import React from 'react';
import {withRouter} from 'react-router';
import CvBlock from './CvBlock';
import {createStyle} from '../modules/utils';
import {loadFirebase} from './withDataLoad';

//отображение документ по ссылке
class CvLink extends React.PureComponent {

    state = {
        blocks: [],
        stylePage: null,
    }

    //загрузим данные по ссылке из базы при монтировании
    componentDidMount() {
        let linkName = this.props.match.params.linkname;
        if (linkName) {
            this.loadLink(linkName);
        }
    }

    //загрузка из firebase
    loadLink = async (linkName) => {
        let loadLinkData = new Promise((resolve) => {
            loadFirebase('Links',linkName,resolve);
        });
        await loadLinkData.then((data) => {
            this.setState({blocks: data.blocks,stylePage: data.style});
        });
    }

    render () {
        if (!this.state.blocks) {
            return null;
        }
        var cvBlocksCode = this.state.blocks.map( b => {
            return <CvBlock key={b.id} id={b.id} data={b} editable={false}></CvBlock>;
        });

        return (
            <React.Fragment>
                <main className='main'>
                    <div className='cv-view cv-view-print'>
                        <div className='cv cv--link' style={createStyle(this.state.stylePage)}>
                            {cvBlocksCode}
                        </div>
                    </div>
                </main>
            </React.Fragment>
        );
    }
}

export default withRouter(CvLink);
