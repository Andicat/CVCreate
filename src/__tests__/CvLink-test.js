

import {renderComponentWithStore, storeWithData} from './utils-test';

import CvLink from '../components/CvLink';

let component;

let props = {match: {params: {linkname: '111'}}};

describe('CvLink render', () => {

    it('Рендер', () => {
        component = renderComponentWithStore(CvLink,storeWithData,props);
        expect(component.toJSON()).toMatchSnapshot();
    });
});

