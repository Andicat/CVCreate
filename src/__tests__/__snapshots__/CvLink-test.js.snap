// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CvLink render Рендер 1`] = `
<main
  className="main"
>
  <div
    className="cv-view cv-view-print"
  >
    <div
      className="cv cv--link"
      style={Object {}}
    />
  </div>
</main>
`;
